flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -187,6 +187,18 @@ public static void setGenerationAnnotation(Configuration conf, Long generation)
         conf.set(KubernetesConfigOptions.JOB_MANAGER_ANNOTATIONS, labels);
     }
 
+    public static void setTaskmanagerGenerationAnnotation(Configuration conf, Long generation) {
+        if (generation == null) {
+            return;
+        }
+        var labels =
+                new HashMap<>(
+                        conf.getOptional(KubernetesConfigOptions.TASK_MANAGER_ANNOTATIONS)
+                                .orElse(Collections.emptyMap()));
+        labels.put(CR_GENERATION_LABEL, generation.toString());
+        conf.set(KubernetesConfigOptions.TASK_MANAGER_ANNOTATIONS, labels);
+    }
```

Comment:
I feel this method duplicates too much code, we could simply have a generic method that takes the TASK_MANAGER/JOBMANAGER_ANNOTATIONS config option as param to factor out the logic

Reply:
I have taken the TASK_MANAGER/JOBMANAGER_ANNOTATIONS config option as param to refactor annotation generation